### PR TITLE
monitoring: do not allow interval variables in alerts

### PR DESCRIPTION
**Pull request by bobheadxi**
tl;dr: with this change we now forbid the use of Grafana magic in `NoAlert: false` graphs - no existing observables are affected by this.

We have a mechanism today, `promql.VariableApplier`, that needs to be able to convert a PromQL query with Grafana variables, to a valid PromQL query _without_ Grafana variables, and then back to the original query again. This is used for label injection. For the reversion to work with numeric values without accidentally converting legitimate values, we "scramble" the numbers somewhat. This works poorly with alert queries, because we do not want a random interval on the alert query.

For now, we isolate the use cases with a new `promql.VariableApplier` constructor that omits interval variables when building alert queries. This will cause `Observable`s that use interval variables _and_ configure an alert to fail on generation. It appears we don't have any such `Observables`s at the moment, but this guards against accidental usage.

A potential future follow-up could be to design a mechanism such that alerts can configure values for interval variables, including `$__rate_interval`, that the alert can use when rendering the final PromQL alert query.

This change was prompted by [this conversation in Slack about adopting `$__rate_interval` more widely](https://sourcegraph.slack.com/archives/C07KZF47K/p1671028821586699).

## Test plan

`go generate ./monitoring` and unit test

### CODE REVIEW

**Review by keegancsmith — APPROVED** [review]
So this means in a NoAlert graph we could now use grafana magic?
